src/include/migraphx/op/reduce_op.hpp

```diff
@@ -107,16 +108,40 @@ struct reduce_op : op_name<Derived>
 
     shape normalize_compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(1);
-        auto s          = inputs.at(0);
-        auto lens       = s.lens();
-        auto tuned_axes = tune_axes(lens.size());
-        for(auto axis : tuned_axes)
+        check_shapes{inputs, *this, true}.has(1);
+        auto s = inputs.at(0);
+        if(s.dynamic())
         {
-            lens[axis] = 1;
+            std::vector<shape::dynamic_dimension> output_dyn_dims = {};
+
+            // create a dynamic dimensions vector that leaves out any axis named in this->axes.
```

Comment:
We shouldn't leave out the axes. The output should be the same rank as the input.

Reply:
I assume you would do:

```cpp
auto dyn_dims = s.dyn_dims();
for(auto axis : tuned_axes)
{
    dyn_dims[axis] = {1, 1};
}
```

Reply:
The onnx specification allows for both, with a flag to choose.  Are we sure we want to keep the axis (and set it to {1,1,1})?

Reply:
The current behavior for static shapes always keeps the dimensions, and the dynamic version should work the same. We dont support removing the dimensions.

Our operators are not the same as onnx operators. For ONNX, we insert a `squeeze` when `keepdims` is false: https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/onnx/parse_reduce_op.cpp#L90